src/Form/AuthenticationSettingsForm.php

```diff
@@ -184,17 +184,28 @@ protected function getUserConditionDefinitions(): array {
     $definitions = $this->conditionManager->getFilteredDefinitions('oe_authentication_2fa', [$context]);
     // The ::getFilteredDefinitions() call above filters out plugins that match
     // the context definition, as well as plugins without any context definition
-    // required. We need to filter those out manually.
-    return array_filter($definitions, function ($definition) {
+    // required, and plugins with optional context definitions.
+    // We need to filter those out manually, and return only conditions that
+    // require one single user context.
+    return array_filter($definitions, function ($definition): bool {
       // @see \Drupal\Core\Plugin\Context\ContextHandler::getContextDefinitions()
       if ($definition instanceof ContextAwarePluginDefinitionInterface) {
-        return !empty($definition->getContextDefinitions());
+        $context_definitions = $definition->getContextDefinitions();
       }
-      if (is_array($definition) && isset($definition['context_definitions'])) {
-        return !empty($definition['context_definitions']);
+      elseif (is_array($definition) && isset($definition['context_definitions'])) {
+        $context_definitions = $definition['context_definitions'];
```

Comment:
We _could_ do this:
```suggestion
      elseif (is_array($definition)) {
        $context_definitions = $definition['context_definitions'] ?? [];
```
We already check for `empty()` below.
But, there is some symmetry to skipping those without a context definition, it is symmetric to the instanceof above.